src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionTemplateService.kt

```diff
@@ -482,6 +501,21 @@ class SessionTemplateService(
 
     return overlappingSessions
   }
+
+  fun moveSessionTemplateVisits(fromSessionTemplateReference: String, toSessionTemplateReference: String, fromDate: LocalDate): Int {
+    val fromSessionTemplate = SessionTemplateDto(getSessionTemplate(fromSessionTemplateReference))
+    val toSessionTemplate = SessionTemplateDto(getSessionTemplate(toSessionTemplateReference))
+    val fromSessionTemplateVisitStats = getSessionTemplateVisitStats(fromSessionTemplate.reference, RequestSessionTemplateVisitStatsDto(fromDate))
+    val toSessionTemplateVisitStats = getSessionTemplateVisitStats(toSessionTemplate.reference, RequestSessionTemplateVisitStatsDto(fromDate))
+
+    visitMoveValidator.validateMoveSessionTemplateVisits(fromSessionTemplate, fromSessionTemplateVisitStats, toSessionTemplate, toSessionTemplateVisitStats, fromDate)
+
```

Comment:
Would move :
```
    val fromSessionTemplateVisitStats = getSessionTemplateVisitStats(fromSessionTemplate.reference, RequestSessionTemplateVisitStatsDto(fromDate))
    val toSessionTemplateVisitStats = getSessionTemplateVisitStats(toSessionTemplate.reference, RequestSessionTemplateVisitStatsDto(fromDate))
```
into `validateMoveSessionTemplateVisits`

so we encapsulate it correctly

Reply:
moved to a new method in SessionTemplateService.